guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitServiceImpl.java

```diff
@@ -89,278 +92,272 @@
+                                                       final String defaultGroupId) {
+
+        return createOrganizationalUnit(name,
+                                        owner,
+                                        defaultGroupId,
+                                        new ArrayList<>());
+    }
 
-        try {
-            configurationService.startBatch();
-            final ConfigGroup groupConfig = configurationFactory.newConfigGroup( ConfigType.ORGANIZATIONAL_UNIT,
-                                                                                 name,
-                                                                                 "" );
-            groupConfig.addConfigItem( configurationFactory.newConfigItem( "owner",
-                                                                           owner ) );
-            String _defaultGroupId = defaultGroupId == null || defaultGroupId.trim().isEmpty() ? getSanitizedDefaultGroupId( name ) : defaultGroupId;
-            groupConfig.addConfigItem( configurationFactory.newConfigItem( "defaultGroupId",
-                                                                           _defaultGroupId ) );
-            groupConfig.addConfigItem( configurationFactory.newConfigItem( "repositories",
-                                                                           new ArrayList<String>() ) );
-            groupConfig.addConfigItem( configurationFactory.newConfigItem( "security:groups",
-                                                                           new ArrayList<String>() ) );
-            configurationService.addConfiguration( groupConfig );
-
-            newOrganizationalUnit = organizationalUnitFactory.newOrganizationalUnit( groupConfig );
-            registeredOrganizationalUnits.put( newOrganizationalUnit.getName(),
-                                               newOrganizationalUnit );
+    @Override
+    public OrganizationalUnit createOrganizationalUnit(final String name,
+                                                       final String owner,
+                                                       final String defaultGroupId,
+                                                       final Collection<Repository> repositories) {
+        if (!authorizationManager.authorize(OrganizationalUnit.RESOURCE_TYPE,
+                                            OrganizationalUnitAction.CREATE,
+                                            sessionInfo.getIdentity())
+                || registeredOrganizationalUnits.containsKey(name)) {
+            return null;
+        }
 
-            return newOrganizationalUnit;
-        } finally {
-            configurationService.endBatch();
-            if ( newOrganizationalUnit != null ) {
-                newOrganizationalUnitEvent.fire( new NewOrganizationalUnitEvent( newOrganizationalUnit, getUserInfo( sessionInfo ) ) );
-            }
+        if (!ValidationUtils.isFileName(name)) {
```

Comment:
@manstis The problem is that the OU name is being used to create the default repository related to it (for now, as this will change when we have 1 repository to 1 project). The repository name must be a valid file name. 

Another option would be replace special characters by, for instance, "-". But this could cause conflicted repository names e.g. between OUs "name*" and "name-".

Do you have another suggestion?